pkg/descheduler/evictions/evictions.go

```diff
@@ -171,6 +171,7 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, opts EvictOptio
 		eventBroadcaster.StartRecordingToSink(&clientcorev1.EventSinkImpl{Interface: pe.client.CoreV1().Events(pod.Namespace)})
 		r := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "sigs.k8s.io.descheduler"})
 		r.Event(pod, v1.EventTypeNormal, "Descheduled", fmt.Sprintf("pod evicted by sigs.k8s.io/descheduler%s", opts.Reason))
+		eventBroadcaster.Shutdown()
```

Comment:
I wonder whether we actually broadcast the events properly. Checking the KS code [1]:
```
cc.EventBroadcaster.StartRecordingToSink(ctx.Done())
defer cc.EventBroadcaster.Shutdown()
```
and [2]:
```
// Prepare kube clients.
client, eventClient, err := createClients(kubeConfig)
if err != nil {
	return nil, err
}

c.EventBroadcaster = events.NewEventBroadcasterAdapter(eventClient)
```

`StartRecordingToSink` and `Shutdown` is invoked only once.

[1] https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/server.go#L160-L161
[2] https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/options/options.go#L280-L286

Reply:
I would think the eventbroadcaster should be started at startup (followed by a defer), and then just passed everywhere it's used. I agree it seems weird we're starting and stopping it right here

Reply:
From https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/interfaces.go#L42:
```
Eventf(regarding runtime.Object, related runtime.Object, eventtype, reason, action, note string, args ...interface{})
```

The descheduler would then invoke `eventBroadcaster.Eventf(...)`

Reply:
Worth refactoring the code and moving the bits up in https://github.com/kubernetes-sigs/descheduler/blob/d5e66ab62e3683dfce663417ba46f957b5b29b59/pkg/descheduler/descheduler.go#L169. `EvictPod` should be really invoking only a single command to record an event.

Reply:
Looking at it now I see that we probably are dealing with events in a non optimal way here. I will have a proper look at what kube-scheduler is doing there and reproduce it here

Reply:
@knelasevero much appreciate the time and effort in investigating the issue and figuring out where the leak is